test/screenshot.test.js

```diff
@@ -1,36 +1,79 @@
-const fs = require('fs')
-const path = require('path')
-const { env, normalizeName } = require('./config.js')
+/**
+ * Run visual regression tests.
+ *
+ * Note that `npm run test:build` must be run before running these tests.
+ * That script runs Parcel to build the `test/build` directory with all the
+ * HTML, compiled CSS and JS used for visual regression testing in it.
+ */
+
 const assert = require('assert')
+const fs = require('fs')
+const http = require('http')
+const {
+  env: { staticDir, baseUrl, examples, diff }
+} = require('./config.js')
 
-const isHTML = file => file.endsWith('.html')
+// The examples to use for visual regression tests
+// It's generally better to only use small examples, as
+// larger ones take up time and space
+const EXAMPLES = ['article-kitchen-sink']
 
-// Find HTML files stored the examples folder for testing
-const files = fs.readdirSync(env.examples).filter(isHTML)
+// The themes to be tested. Defaults to all
+const THEMES = ['stencila', 'elife', 'nature', 'plos', 'skeleton']
 
-/**
- * Takes an array of visual regression comparison results, and checks if all
- * are within difference tolerances.
- *
- * @param {array {isWithinMisMatchTolerance: boolean}} results
- * @returns boolean
- */
-const allScreenshotsPass = results =>
-  results.reduce(
-    (pass, result) => pass && result.isWithinMisMatchTolerance === true,
-    true
-  )
-
-describe('visual regression: ', () => {
-  files.forEach(function(file) {
-    it(`compare screenshot of ${file}`, async () => {
-      await browser.url('/' + file)
-      const results = await browser.checkDocument()
-
-      assert.ok(
-        allScreenshotsPass(results),
-        `Styles differ from current references. Please see ${env.diff} for differences`
-      )
+// Get hostname and port for doing connectivity tests
+const { hostname, port } = new URL(baseUrl)
+
+describe('visual regressions: ', () => {
+  it(`needs the build folder`, () => {
+    assert.ok(
+      fs.existsSync(staticDir),
+      `${staticDir} does not exist. Did you run 'npm run test:build' first?`
+    )
+  })
+
+  it(`demo page can be got`, () => {
+    const req = http.get({ hostname, port }, res => {
+      assert.equal(res.statusCode, '200')
+    })
+    req.on('error', error => assert.fail(error))
+    req.end()
+  })
```

Comment:
This checks that the host name & port that the tests will run on are available 👍  Might be desirable go more granular and verify that all URIs that will be used for visual regression testing are available? Otherwise could still get 404s getting into the reference screenshots. Taking the approach used in the subsequent test, it might look something like:
``` JavaScript
  it(`demo pages are available`, () => {
    EXAMPLES.forEach(example => {
      THEMES.forEach(theme => {
        const path = `/?example=${example}&theme=${theme}&header=false`
        const req = http.get({ hostname, port, path }, res => {
          assert.equal(res.statusCode, '200')
          req.on('error', error => assert.fail(error))
          })
      });
    })
  })
```
(`req.end()` omitted as this is [called automatically by `http.get()`](https://nodejs.org/dist/latest-v12.x/docs/api/http.html#http_http_get_options_callback))

Reply:
Thanks. I did have `path` in there but it inadvertently got dropped - thanks for catching that - it is the whole point of the check :smile: